src/main/java/servlets/VendorServlet.java

```diff
@@ -33,12 +44,68 @@
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     JsonConverter converter = new JsonConverter();
     String configMap = converter.getConfigMap();
-    response.setContentType("text/html;");
+    response.setContentType(CONTENT_TYPE);
     response.getWriter().println(configMap);
   }
 
-  @Override
-  public void doDelete(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // DELETE method --> deletes vendor configuration directory
+  /** Delete the desired Vendor from the fileset and spreadsheets if posible.*/
+  public void doDelete(HttpServletRequest request, HttpServletResponse response) throws IOException {  
+    String vendorID = request.getParameter(VENDOR_ID_PARAM);
+    String deleteStatus = "success";
+
+    // Delete the file from the filesystem and update the sheets.
+    try {
+      deleteVendorFile(vendorID);
+      updateSheets(request);    
+    } catch (IOException e) {
+      deleteStatus = "Could not execute, an IOException was thrown.";
+    } catch (GeneralSecurityException e) {
+      deleteStatus = "Could not execute, a GeneralSecurityException was thrown.";
+    } 
+
+    // Convert the String message into a JSON String.
+    String jsonMessage = messageAsJson(deleteStatus);
+    
+    response.setContentType(CONTENT_TYPE);
+    response.getWriter().println(jsonMessage);
+  }
+
+  private void deleteVendorFile(String vendorID) throws IOException {
+    JsonConverter converter = new JsonConverter();
+    ArrayList<String> vendorIDs = converter.getVendorIDs();
+
+    // Validate that the vendorID exists.
+    if(!vendorIDs.contains(vendorID)) {
+        throw new IOException();
+    }
+    converter.deleteFile(vendorID);
+  }
+
+  /** Rebuild the sheets without the deleted Vendor's data. */
+  private void updateSheets(HttpServletRequest request) 
+      throws IOException, GeneralSecurityException {
+    JsonConverter converter = new JsonConverter();
+    ArrayList<String> vendorIDs = converter.getVendorIDs();
+    ArrayList<Vendor> vendors = new ArrayList<Vendor>(); 
+
+    // Add the Vendors associated with the existing vendorIDs.
+    for(int i = 0; i < vendorIDs.size(); i++) {
+      String vendorJson = converter.getConfig(vendorIDs.get(i));
+      vendors.add(new Vendor(vendorJson, vendorIDs.get(i)));
+    }
+
```

Comment:
It might be useful to separate this into its own method `getVendors()`, lmk what you think

Reply:
Sure, that would make the updateSheets() method a bit cleaner.